src/battle-log.ts

```diff
@@ -133,11 +139,23 @@ class BattleLog {
 
 		case 'name': case 'n': {
 			const user = BattleTextParser.parseNameParts(args[1]);
-			if (toID(args[2]) !== toID(user.name)) {
-				divHTML = '<small>' + BattleLog.escapeHTML(user.group + user.name) + ' renamed from ' + BattleLog.escapeHTML(args[2]) + '.</small>';
+			if (toID(args[2]) === toID(user.name)) return;
+			if (!this.lastRename) {
+				this.lastRename = {
+					from: args[2],
+					to: '',
+					element: document.createElement('div'),
+				};
+				this.lastRename.element.className = 'chat';
+			} else if (toID(this.lastRename.to) !== toID(user.name)) {
+				this.lastRename.from = args[2];
```

Comment:
```suggestion
			if (!this.lastRename || toID(this.lastRename.to) !== toID(user.name)) {
				this.lastRename = {
					from: args[2],
					to: '',
					element: document.createElement('div'),
				};
				this.lastRename.element.className = 'chat';
```

This seems like what you'd want? Unless you intentionally wanted to hide earlier renames if multiple people renamed in a row.

Reply:
I suppose it wouldn't be horrible if you did, but I don't know if that's necessary.

Reply:
Mm, I suppose multiple users renaming would be pretty difficult to abuse with a bot.